sqlglot/generator.py

```diff
@@ -103,7 +103,7 @@ class Generator:
         exp.TableFormatProperty,
     }
 
-    WITH_SEPARATED_COMMENTS = (exp.Select,)
+    WITH_SEPARATED_COMMENTS = (exp.Select, exp.From, exp.Where, exp.Binary)
```

Comment:
why did you add binary here

Reply:
given ``1=/*c*/2`` and ``1+/*c*/2``, this will result in 

``/* c */ 1 = 2`` and ``/* c */ 1 + 2``

otherwise the comment will be after the expression

Reply:
don’t you want to just keep the formatting?

Reply:
or is the current behavior not 1 + /* x */ 2

Reply:
As I said in my first post, by default the comment would be added after the expression. 

Since ``1 = 2`` is the (equality) expression, the comment would be added afterwards. Of course, keeping the original position of the comment would be preferred, but that's probably to be solved at a later point. As GeorgeSittas sugested, it might be preferrable for some expressions, to have the comment moved to the front, if the alternative is to have it placed at the end.

Reply:
so why is /* x */ 1 + 2 better than 1 + 2 /* x */, should we just leave it the default?

Reply:
Consider this expression:  ``2 * /*c1*/ (3+ /*c2*/ 4)``

Default (comment at the end): ``2 * (3 + 4 /* c2 */) /* c1 */``

Comment at the start: ``/* c1 */ 2 * (/* c2 */ 3 + 4)``

Thus I would argue that the comment at the start is better in this case.